src/main/java/io/confluent/connect/hdfs/TopicPartitionWriter.java

```diff
@@ -398,10 +417,20 @@ private void setState(State state) {
     this.state = state;
   }
 
+  private void setIsFirst(boolean isFirst) {
+    this.isFirst = isFirst;
+  }
+
   private boolean shouldRotate(long now) {
-    boolean periodicRotation = rotateIntervalMs > 0 && now - lastRotate >= rotateIntervalMs;
+    boolean periodicRotation = rotateIntervalMs > 0 && (!isFirst && now - lastRotate >= rotateIntervalMs);
     boolean scheduledRotation = rotateScheduleIntervalMs > 0 && now >= nextScheduledRotate;
     boolean messageSizeRotation = recordCounter >= flushSize;
+
+    if (isFirst) {
```

Comment:
Seems like we could also avoid the whole `isFirst` business if we just initialize `lastRotate` to be the current time in the constructor.